nvm.sh

```diff
@@ -2230,7 +2231,7 @@ nvm_die_on_prefix() {
   local NVM_OS
   NVM_OS="$(nvm_get_os)"
   NVM_NPM_PREFIX="$(npm config --loglevel=warn get prefix)"
-  if [ "${NVM_VERSION_DIR}" != "${NVM_NPM_PREFIX}" ] && ! (nvm_tree_contains_path "${NVM_VERSION_DIR}" "${NVM_NPM_PREFIX}" >/dev/null 2>&1); then
+  if [ ! "${NVM_VERSION_DIR}" -ef "${NVM_NPM_PREFIX}" ] && ! (nvm_tree_contains_path "${NVM_VERSION_DIR}" "${NVM_NPM_PREFIX}" >/dev/null 2>&1); then
```

Comment:
this will need tests that fail without the change - NVM_VERSION_DIR should never have a trailing slash in the first place (directories don't have slashes; the slash means the *contents* of the directory), and i'm not sure what else `-ef` does (do you have docs for it that demonstrate it's in posix and describe its semantics?

Reply:
https://tldp.org/LDP/Bash-Beginners-Guide/html/sect_07_01.html
https://tldp.org/LDP/abs/html/fto.html

I can't find when it was created in any documentation, but based on stackexchange threads, it's fully compatible with all shells. 

It checks if 2 files refer to the same device and inode numbers.

This can be tested manually in all posix-shells with:
```
# Prints
test "$PWD" -ef . && echo 'hello'
# Doesn't print
test "$PWD/../" -ef . && echo 'hello'
```

Reply:
Thanks, this seems useful. There's probably other places that could use this change, like `nvm_tree_contains_path`, and `nvm_find_up` etc.